### ISSUE TITLE
Implement `GET /applications` endpoint

### ISSUE THREAD

**Issue by slifty** [PhilanthropyDataCommons/service#34]
Returns a list of all applications int he system.

This should support an optional `opportunityId` querystring parameter which filters the returned applications by opportunityId.

**Comment by bickelj** [comment]
Depends on #64 and #65

**Comment by slifty** [comment]
I'm going to create a separate issue for the `/proposals/{proposalId}` endpoint.
